prow/cmd/checkconfig/main.go

```diff
@@ -1052,23 +1052,45 @@ func verifyOwnersPlugin(cfg *plugins.Configuration) error {
 	return nil
 }
 
+func checkRestrictedLabelNotInAdditionalLables(restricted string, label plugins.Label) bool {
+	for _, additional := range label.AdditionalLabels {
+		if restricted == additional {
+			return true
+		}
+	}
+	return false
+}
+
 func verifyLabelPlugin(label plugins.Label) error {
 	var orgRepos []string
+	var errs []error
+	misconfiguredLabels := make(map[string][]string)
 	for orgRepo, restrictedLabels := range label.RestrictedLabels {
 		for _, restrictedLabel := range restrictedLabels {
+			if found := checkRestrictedLabelNotInAdditionalLables(restrictedLabel.Label, label); found {
+				misconfiguredLabels[restrictedLabel.Label] = append(misconfiguredLabels[restrictedLabel.Label], orgRepo)
+			}
 			if restrictedLabel.Label == "" {
 				orgRepos = append(orgRepos, orgRepo)
 			}
 		}
 	}
 
+	if len(misconfiguredLabels) > 0 {
```

Comment:
no need for this check here, since you `make` the map above, the map won't be nil, and if it's empty the interaction won't happen.